CodeSnippetsReflection.OpenAPI/LanguageGenerators/SnippetImports.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using CodeSnippetsReflection.OpenAPI.ModelGraph;
+using CodeSnippetsReflection.StringExtensions;
+using Microsoft.OpenApi.Services;
+
+namespace CodeSnippetsReflection.OpenAPI.LanguageGenerators;
+
+public class ImportsGenerator
+{
+    public List<Dictionary<string, string>> imports = new();
+    public List<Dictionary<string, string>> GenerateImportTemplates(SnippetModel snippetModel)
+    {
+        var codeGraph = new SnippetCodeGraph(snippetModel);
+        // Call recursive function to get all model import to be added to the template
+        var imports = new List<Dictionary<string, string>>();
+        if (codeGraph.HasHeaders() || codeGraph.HasParameters() || codeGraph.HasOptions())
+        {
+            // we have a request builder
+            var className = codeGraph.Nodes.Last().GetClassName().ToFirstCharacterUpperCase();
+            var itemSuffix = codeGraph.Nodes.Last().Segment.IsCollectionIndex() ? "Item" : string.Empty;
+            var requestBuilderName = $"{className}{itemSuffix}RequestBuilder";
+            //check path and include last part to request builder name
+            if (codeGraph.Nodes.Last().Path != null)
+            {
+                var path = codeGraph.Nodes.Last().Path;
+                var pathParts = path.Split('/');
+                var lastPathPart = pathParts.Last();
+                if (lastPathPart != null)
+                {
+                    requestBuilderName = $"{requestBuilderName}";
+                    imports.Add(new Dictionary<string, string>
+                    {
+                        { "Path", Regex.Replace(path.Replace("\\", ".").Replace("()", ""), @"\{[^}]*-id\}", "item", RegexOptions.None, TimeSpan.FromSeconds(60))},
```

Comment:
We should probably create a static class property(so we don't keep instantiating a new one) for the regex and pass the compile flag too. 
Similar to https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/723913bef8fed8a109697ec51e367ea5d3c2f1fb/TelemetryService/CustomPIIFilter.cs#L30
```suggestion
                        { "Path", Regex.Replace(path.Replace("\\", ".").Replace("()", ""), @"\{[^}]*-id\}", "item", RegexOptions.Compiled, TimeSpan.FromSeconds(60))},
```

Reply:
This is a one off instance, its not used elsewhere, would that still require a static class?